src/TensorFlowNET.Core/Eager/c_api.eager.cs

```diff
@@ -65,7 +66,7 @@ public partial class c_api
         /// <param name="status">TF_Status*</param>
         /// <returns></returns>
         [DllImport(TensorFlowLibName)]
-        public static extern int TFE_OpAddInputList(SafeOpHandle op, IntPtr[] inputs, int num_inputs, SafeStatusHandle status);
+        public static extern int TFE_OpAddInputList(SafeOpHandle op, [In, MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(SafeHandleArrayMarshaler))] SafeTensorHandleHandle[] inputs, int num_inputs, SafeStatusHandle status);
```

Comment:
📝 `SafeHandleArrayMarshaler` works for passing arrays of safe handles _to_ unmanaged code, but not the other way around. The `[In]` attribute specifies that I only want to send data one way.

Reply:
Are you saying we can't passing array from unmanged code to .NET ?

Reply:
We can, but not with the automatic marshaling. See how the `TFE_Execute` call is wrapped for an example.

Reply:
```csharp
            var outputs = new IntPtr[num_outputs];
            if (status.ok())
            {
                c_api.TFE_Execute(op, outputs, ref num_outputs, status.Handle);
                status.Check(true);
            }
```
Can the `outputs ` be Safe Handles `TFE_TensorHandle[]` ?

Reply:
Are you referring to code like this?

https://github.com/SciSharp/TensorFlow.NET/blob/c9f268cdcfd1191c9c92ecbdd59741336f3e391a/test/TensorFlowNET.UnitTest/NativeAPI/Eager/CApi.Eager.OpInferMixedTypeInputListAttrs.cs#L45-L50

Reply:
https://github.com/SciSharp/TensorFlow.NET/blob/81215df6e6397d5dbb79ecaa50ed07e997851494/src/TensorFlowNET.Core/Eager/EagerRunner.TFE_Execute.cs#L56

Reply:
I just saw you have updated the code. So never mind.